File: Frontend/src/types/notification.ts
export enum NotificationChannel {
  None = 0,
  SMS = 1,
  Email = 2,
  Push = 3,
  InApp = 4,
}

export enum NotificationPriority {
  Low = 1,
  Normal = 2,
  High = 3,
  Critical = 4,
}

export enum NotificationStatus {
  Pending = 1,
  Sent = 2,
  Failed = 3,
  Cancelled = 4,
  IsRead = 5,
}

export enum NotificationType {
  AppointmentConfirmation = 1,
  AppointmentReminder = 2,
  AppointmentCancellation = 3,
  PrescriptionReady = 4,
  TestResultsAvailable = 5,
  PaymentDue = 6,
  PaymentConfirmation = 7,
  AppointmentCancelledByDoctor = 8,
  FactureAdded = 9,
  AppointmentCreated = 10,
  AppointmentUpdated = 11,
  NewAppointment = 20,
  AppointmentModified = 21,
  PatientArrived = 22,
  UrgentMessage = 23,
  TestResultsReceived = 24,
  NewPatientRegistered = 40,
  LastMinuteCancellation = 41,
  PaymentIssue = 42,
  StockAlert = 43,
  EquipmentMaintenance = 44,
  SystemAlert = 60,
  SecurityAlert = 61,
  BackupFailed = 62,
  PerformanceReport = 63,
}

export enum UserType {
  Patient = 1,
  Doctor = 2,
  ClinicAdmin = 3,
  SuperAdmin = 4,
}

export interface CreateNotificationRequest {
  recipientId: string;
  recipientType: UserType;
  type: NotificationType;
  title: string;
  content: string;
  priority?: NotificationPriority;
  scheduledAt?: string | null;
  metadata?: Record<string, unknown> | null;
}

export interface MarkNotificationAsSentRequest {
  notificationId: string;
  sentAt: string;
}

export interface NotificationDto {
  id: string;
  recipientId: string;
  recipientType: UserType;
  type: NotificationType;
  channel: NotificationChannel;
  title: string;
  content: string;
  priority: NotificationPriority;
  status: NotificationStatus;
  createdAt: string;
  sentAt?: string | null;
}

export interface NotificationSummaryDto {
  id: string;
  title: string;
  content: string;
  status: NotificationStatus;
  createdAt: string;
}

export interface NotificationFilterRequest {
  recipientId?: string | null;
  recipientType?: UserType | null;
  status?: NotificationStatus | null;
  type?: NotificationType | null;
  from?: string | null;
  to?: string | null;
  page?: number;
  pageSize?: number;
}

export interface SendNotificationRequest {
  notificationId: string;
}

File: Services/ReportingService/Reporting.Domain/Entities/Notification.cs
using Notif.Domain.Enums;

namespace Notif.Domain.Entities
{
    public class Notification
    {
        public Guid Id { get; set; }
        public Guid RecipientId { get; set; }
        public UserType RecipientType { get; set; }
        public NotificationType Type { get; set; }
        public NotificationChannel Channel { get; set; }
        public string Title { get; set; }
        public string Content { get; set; }
        public NotificationPriority Priority { get; set; }
        public NotificationStatus Status { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? ScheduledAt { get; set; }
        public DateTime? SentAt { get; set; }
        public int RetryCount { get; set; }
        public string? ErrorMessage { get; set; }

        public static Notification Create(
            Guid recipientId,
            UserType recipientType,
            NotificationType type,
            NotificationChannel channel,
            string title,
            string content,
            NotificationPriority priority = NotificationPriority.Normal,
            DateTime? scheduledAt = null)
        {
            return new Notification
            {
                Id = Guid.NewGuid(),
                RecipientId = recipientId,
                RecipientType = recipientType,
                Type = type,
                Channel = channel,
                Title = title,
                Content = content,
                Priority = priority,
                Status = NotificationStatus.Pending,
                CreatedAt = DateTime.UtcNow,
                ScheduledAt = scheduledAt,
                RetryCount = 0,
            };
        }

        public void MarkAsSent()
        {
            Status = NotificationStatus.Sent;
            SentAt = DateTime.UtcNow;
        }

        public void MarkAsFailed(string errorMessage)
        {
            Status = NotificationStatus.Failed;
            ErrorMessage = errorMessage;
            RetryCount++;
        }

        public void Retry()
        {
            Status = NotificationStatus.Pending;
            ErrorMessage = null;
        }

        public bool CanRetry() => RetryCount < 3 && Status == NotificationStatus.Failed;
    }
}
